src/FilesystemRepository.php

```diff
@@ -124,10 +119,10 @@ public static function isSymlinkSupported()
      */
     public function __construct($baseDir = '/', $symlink = true, $relative = true)
     {
-        Assert::directory($baseDir);
+        parent::__construct($baseDir);
+
         Assert::boolean($symlink);
 
-        $this->baseDir = rtrim(Path::canonicalize($baseDir), '/');
```

Comment:
Using the same directory does not make sense at all. By default, for example the directory where FilesystemRepository stores its resources is `<root>/.puli/repository`, while the base directory (for security checks) is `<root>`.

I think we should remove the security checks again. Let's KISS:

* PathMappingRepository stores its paths relative to `$baseDir`
* FilesystemRepository stores its content in `$baseDir`